src/distro/distro.py

```diff
@@ -1089,6 +1095,59 @@ def uname_attr(self, attribute: str) -> str:
         """
         return self._uname_info.get(attribute, "")
 
+    @staticmethod
+    def __abs_path_join(root_path: str, abs_path: str) -> str:
+        rel_path = os.path.splitdrive(abs_path)[1].lstrip(os.sep)
+        if os.altsep is not None:
+            rel_path = rel_path.lstrip(os.altsep)
+
+        return os.path.join(root_path, rel_path)
+
+    def __resolve_chroot_symlink_as_needed(self, link_location: str) -> str:
+        """
+        Resolves a potential symlink in ``link_location``
+        against ``self.root_dir`` if inside the chroot,
+        else just return the original path.
+        We're doing this check at a central place, to making
+        the calling code more readable and to de-duplicate.
+        """
+        if self.root_dir is None:
+            return link_location
+
+        if os.path.commonprefix([self.root_dir, link_location]) != self.root_dir:
+            raise FileNotFoundError
```

Comment:
I believe we need…
- to move this code into the `While True` loop below, and
- to use `os.path.commonpath` (not `commonprefix`) of Python >=3.5 instead OR keep using `os.path.commonprefix` but with a copy of `self.root_dir` ending in exactly one trailing instance of `os.sep`, in order to not run into false positives.